controllers/dynamic_resource_request_controller.go

```diff
@@ -103,30 +98,36 @@ func (r *dynamicResourceRequestController) Reconcile(ctx context.Context, req ct
 		return addFinalizers(ctx, r.Client, rr, []string{workFinalizer, workflowsFinalizer, deleteWorkflowsFinalizer}, logger)
 	}
 
-	if !r.hasCondition(PipelineCompletedCondition, rr) {
-		r.setStatus(rr, logger, "message", "Pending")
-		r.setCondition(clusterv1.Condition{
-			Type:               PipelineCompletedCondition,
-			Status:             v1.ConditionFalse,
-			Message:            "Pipeline has not completed",
-			Reason:             "PipelineNotCompleted",
-			LastTransitionTime: metav1.NewTime(time.Now()),
-		}, rr, logger)
-		return ctrl.Result{}, r.Client.Status().Update(ctx, rr)
+	pipelineJobs, err := r.getPipelineJobs(resourceRequestIdentifier, rr.GetNamespace(), logger)
+	if err != nil {
+		logger.Info("Failed getting pipeline jobs", "error", err)
+		return slowRequeue, nil
 	}
 
-	//check if the pipeline has already been created. If it has exit out. All the lines
-	//below this call are for the one-time creation of the pipeline.
-	created, err := r.configurePipelineHasBeenCreated(resourceRequestIdentifier, rr.GetNamespace(), logger)
-	if err != nil {
-		return ctrl.Result{}, err
+	if len(pipelineJobs) != 0 {
+		if resourceutil.IsThereAPipelineRunning(logger, pipelineJobs) {
+			return slowRequeue, nil
+		}
+
+		if !resourceutil.IsAnUpdate(logger, rr, pipelineJobs) {
```

Comment:
nitpick: we only call this in the context of checking the negative rather than the positive, maybe inverse the func name?

I'm also struggling to understand what the function does from the name 🤔  is it more `isLastDeployedJobUsingCurrentSpec` (but better)

Reply:
Agreed it would help clarify with a slightly different name. Another option could be `RequiresAnUpdate`.

Reply:
my reasoning behind the name of this functions were based on the package `resourceutil`. From a resource request point of view, what we are trying to determine whether:

- is there a pipeline running for the resource request?
- is this resource request an update?

You are both right that the suggested names better explain what the function is _doing_, but i'd say they are worse in explaining _why_.

Reply:
> is this resource request an update?

I think I read `resourceutil` to be k8s resources, which I'm only now grokking as resource (request) utils which adds a lot more clarity to me, but `IsAnUpdate` still sounds odd, the negative checking might just be a big nit on my side. 

> You are both right that the suggested names better explain what the function is doing, but i'd say they are worse in explaining why.

🤔 this is a super interesting comment! I think knowing what the function is doing from a function name is a good outcome, it saves the caller from reading the function. But maybe thats more suitable for local functions, and less for a library thats trying to encapsulate the logic (I would still says its not a bad outcome for a library though!). I think using the function name to describe why is super interesting here, but I still think `!resourceutil.IsAnUpdate(logger, rr, pipelineJobs)` hard to read 😅

Reply:
Maybe this function should be in the `pipeline` package? Since its searching through the jobs created by the package it seems reasonable, then @abangser's `RequiresAnUpdate` feels cleaner as `pipeline.RequiresAnUpdate`?

Reply:
>I think using the function name to describe why is super interesting here

I think it's preferable!
If tomorrow we decide to determine whether is an update based on a label, or include metadata, or something, we don't need to update the function name... lemme go back to the board and try something considering your feedback 🤔